server/model/status_page.js

```diff
@@ -61,6 +62,26 @@ class StatusPage extends BeanModel {
         }
     }
 
+    /**
+     * Generate SVG Status
+     * @param {response} response express response
+     * @param {slug} slug from router
+     * @param {config} config config for the svg
+     * @returns {Promise<string>} returns a svg file
+     */
+    static async handleStatusPageSVGResponse(response, slug, config) {
+        slug = slug.replace(".svg", "");
+        let statusPage = await R.findOne("status_page", " slug = ? ", [
+            slug.replace(".svg", "")
+        ]);
+        if (statusPage) {
+            response.set("Content-Type", "image/svg+xml");
+            response.send(await StatusPage.renderSVG(slug, config, statusPage));
+        } else {
+            response.status(404).send(await StatusPage.renderSVG(slug, config, null));
```

Comment:
In cases of 404s, aborting early is usually a good call.

aborting here also has downstream effects in the code of `renderSVG` and upstream in the html

```suggestion
            response.status(404).send(UptimeKumaServer.getInstance().indexHTML);
```